New package: StableHLO v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: StableHLO
- Repository: https://github.com/mofeing/StableHLO.jl
- Created by: @mofeing
- Version: v0.1.0
- Commit: 40c0e4f1d3181d6946399f3b219cd93e26f64788
- Git reference: HEAD
- Description: Julia bindings for HLO dialects

goerz commented:
Thank you for submitting your package! However, please make sure to add some documentation before registering. [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a README with a description of the package's purpose and a small usage example.

Without a README, it's not even clear what "HLO" might refer to, and whether `StableHLO` is a suitable package name.

mofeing commented:
This is a common package for the MLIR.jl, Coil.jl and Reactant.jl packages. Since the whole MLIR ecosystem with Julia is still experimental, I haven't added a documentation and users should not directly use this code yet.

goerz commented:
If we're going to hold all registrations to the same standard (which I very much think I should), there is no package that should be registered without a README. It's not a high bar to have one or two paragraphs that explain the purpose of the package and how it should be used (or maybe not used, if only serving as a dependency). If you flesh out [your comment](https://github.com/JuliaRegistries/General/pull/108874#issuecomment-2166908476) just a little bit more, that's all the information that needs to go in the README, as a start.

On the other hand…

> but it's not ready for general usage.

But that's the explicit prerequisite for having a package in General! When you put it like that, it very much sounds like this package should not be registered (yet). If the MLIR ecosystem is still highly experimental, the best thing to do would probably be to have a LocalRegistry for any interdependent packages in that domain, for now.

I'm not actually opposed to having this in General, but it does need to meet the minimum requirements: An average Julia user coming across it must be able to figure out what this package is for, i.e., it must have a README to give some context.

> we are working hard

I understand that, but I think we should avoid any kind of "move fast and break things" in General, even (and maybe, especially) for very technical packages and experienced Julia contributors. It also sets a bad precedent when we don't consistently apply the guidelines for registrations. So please spend a few minutes to add a README.

I'm also not enthusiastic about the name (HLO is not a commonly understood acronym), but I can't really come up with something better… So I'll let that go (but maybe others have some ideas).

[noblock]

kellertuer commented:
Even `StableHighLevelOperations` as a name of the package would – for me – be too vague to know what the package is actually about. HLO itself is definitely not commonly understood as an acronym, I agree with @goerz on that.

Besides that – yes please add at least a Readme, and I also agree that a package that should not (yet) be used, is maybe not suitable for the General Registry? What is the reason to register it, when it is not yet usable? For you own development you could set up your own local registry.

vchuravy commented:
It's the Julia wrapper for https://github.com/openxla/stablehlo and thus falls under the same category as MLIR/LLVM/FFMPEG/... 

I agree that a Readme is needed, but the name is fine.

goerz commented:
Thanks for clarifying that! I agree that the name is fine, then.

> It's the Julia wrapper for https://github.com/openxla/stablehlo

That would be one of the things to put in the README, of course. (If it had been, I wouldn't have had to raise any issue about the name 😉)

[noblock]

JuliaHubRegistrator committed:
New package: StableHLO v0.1.0

UUID: 11df7cf1-74a9-4cb7-b537-510301b49c1b
Repo: https://github.com/mofeing/StableHLO.jl.git
Tree: 836ae4b99e354bb419f5a01a9f4e4224d0ea3e11

Registrator tree SHA: 191228b6dd8b9d0e2965ae3e705fe54c51dcfee8

mofeing commented:
Okay, just added a small README to the repo and pointed to some link for more info.

[noblock]

goerz commented:
[noblock] That's *very* minimal… Remember that the point of General is for the packages that registered there to be accessible to the general public. Based on that README, do you think the average Julia user would be able to figure out why this exists or what it's used for? I really don't like this disregard for documentation. But I'll let it pass, and unblock my comment. I think @vchuravy's https://github.com/JuliaRegistries/General/pull/108874#issuecomment-2168450082 needs a `[noblock]` to proceed, though.
